File: Java/easy/28. Find the Index of the First Occurrence in a String/28_Find_the_Index_of_the_First_Occurrence_in_String.java
class Solution {
    public int strStr(String haystack, String needle) {
        if (needle.isEmpty()) return 0;
        int len1 = haystack.length();
        int len2 = needle.length();
        for (int i = 0; i <= len1 - len2; i++) {
            if (haystack.substring(i, i + len2).equals(needle)) {
                return i;
            }
        }
        return -1;
    }
}

File: Golang/28. Find the Index of the First Occurrence in a String/28_Find_the_Index_of_the_First_Occurrence_in_String.go
func strStr(haystack string, needle string) int {
    if len(needle) == 0 {
        return 0
    }
    for i := 0; i <= len(haystack)-len(needle); i++ {
        if haystack[i:i+len(needle)] == needle {
            return i
        }
    }
    return -1
}
